kubernetes/istio_details_service.go

```diff
@@ -196,14 +196,27 @@ func (in *IstioClient) getRbacResources() map[string]bool {
 // If serviceName param is provided it will filter all VirtualServices having a host defined on a particular service.
 // It returns an error on any problem.
 func (in *IstioClient) GetVirtualServices(namespace string, serviceName string) ([]IstioObject, error) {
-	result, err := in.istioNetworkingApi.Get().Namespace(namespace).Resource(virtualServices).Do().Get()
-	if err != nil {
-		return nil, err
+	var virtualServiceList *GenericIstioObjectList
+	if in.k8sCache != nil && in.k8sCache.IstioCached(namespace) {
```

Comment:
yeah, here I guess it can also check the (namespace, type), and probably with that tradeoff between caching / performance may work.